src/sql/src/parsers/alter_parser.rs

```diff
@@ -143,12 +148,42 @@ impl ParserContext<'_> {
                 .parse_identifier(false)
                 .context(error::SyntaxSnafu)?,
         );
-        let target_type = self.parser.parse_data_type().context(error::SyntaxSnafu)?;
 
-        Ok(AlterTableOperation::ChangeColumnType {
-            column_name,
-            target_type,
-        })
+        if self.parser.parse_keyword(Keyword::SET) {
+            self.parser
+                .expect_keyword(Keyword::FULLTEXT)
+                .context(error::SyntaxSnafu)?;
+
+            let options = self
+                .parser
+                .parse_options(Keyword::WITH)
+                .context(error::SyntaxSnafu)?
+                .into_iter()
+                .map(parse_option_string)
+                .collect::<Result<HashMap<String, String>>>()?;
+
+            for key in options.keys() {
+                ensure!(
+                    key.to_ascii_lowercase().as_str() == COLUMN_FULLTEXT_CHANGE_OPT_KEY_ENABLE
+                        || validate_column_fulltext_option(key.to_ascii_lowercase().as_str()),
```

Comment:
Maybe we can provide two methods：
 `validate_column_fulltext_create_option` and `validate_column_fulltext_alter_option`

```
fn validate_column_fulltext_alter_option(key) {
return key == COLUMN_FULLTEXT_CHANGE_OPT_KEY_ENABLE || validate_column_fulltext_create_option(key)
}
```